crypto/aria/aria.c

```diff
@@ -1110,6 +1133,12 @@ void aria_encrypt(const unsigned char *in, unsigned char *out,
                   const ARIA_KEY *key)
 {
     assert(in != NULL && out != NULL && key != NULL);
+
+    if (FIPS_mode()) {
+        FIPSerr(ERR_LIB_FIPS, FIPS_R_NON_FIPS_METHOD);
+        return;
```

Comment:
is the the right thing to do here?

They seem to be asserting they are null ... but if we bailed early the data in `*out` will be unmodified and `do_encrypt` is also a void.

Should we be asserting instead and treat it like something is `NULL` in the input parameters?

Reply:
Yes it is the correct thing to do for FIPS. Any time a non-approved function is called, we need to log that fact (that an attempt was made to use a non-approved function). So it's not the same as the NULL input parameter case.

Reply:
asserting did cross my mind, the only other place we asserted is when the program tried to exit FIPS mode.

Reply:
Won't asserting terminate the program ? That would break openssl speed, which tries to instantiate (and check for NULL return) to determine if we need to test the speed of a particular algo.

Calling FIPSerr in these cases are a failure indication that can be read by the calling program.